yt_dlp/extractor/nfb.py

```diff
@@ -14,29 +58,183 @@ class NFBIE(InfoExtractor):
+            'id': 'liste-des-choses-qui-existent-saison-2-ours',
+            'ext': 'mp4',
+            'title': 'La liste des choses qui existent - L\'ours en peluche',
+            'description': 'md5:d5e8d8fc5f3a7385a9cf0f509b37e28a',
+            'series': 'La liste des choses qui existent',
+            'release_year': 2022,
+            'season': 'Saison 2',
+            'season_number': 2,
+            'episode': 'L\'ours en peluche',
+            'episode_number': 12,
+            'uploader': 'Francis Papillon',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+        'params': {'skip_download': 'm3u8'},
+    }, {
+        'note': 'NFB film /embed/player/ page',
+        'url': 'https://www.nfb.ca/film/afterlife/embed/player/',
+        'info_dict': {
+            'id': 'afterlife',
+            'ext': 'mp4',
+            'title': 'Afterlife',
+            'description': 'md5:84951394f594f1fb1e62d9c43242fdf5',
+            'release_year': 1978,
+            'duration': 420.0,
+            'uploader': 'Ishu Patel',
+            'thumbnail': r're:^https?://.*\.jpg$',
         },
+        'params': {'skip_download': 'm3u8'},
     }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-
-        webpage = self._download_webpage('https://www.nfb.ca/film/%s/' % video_id, video_id)
-
-        iframe = self._html_search_regex(
-            r'<[^>]+\bid=["\']player-iframe["\'][^>]*src=["\']([^"\']+)',
-            webpage, 'iframe', default=None, fatal=True)
-        if iframe.startswith('/'):
-            iframe = f'https://www.nfb.ca{iframe}'
-
-        player = self._download_webpage(iframe, video_id)
+        site, type_, slug = self._match_valid_url(url).group('site', 'type', 'id')
+        # Need to construct the URL since we match /embed/player/ URLs as well
+        webpage, urlh = self._download_webpage_handle(f'https://www.{site}.ca/{type_}/{slug}/', slug)
+        # type_ can change from film to serie(s) after redirect; new slug may have episode number
+        type_, slug = self._match_valid_url(urlh.url).group('type', 'id')
 
-        source = self._html_search_regex(
-            r'source:\s*\'([^\']+)',
-            player, 'source', default=None, fatal=True)
+        embed_url = urljoin(f'https://www.{site}.ca', self._html_search_regex(
+            r'<[^>]+\bid=["\']player-iframe["\'][^>]*\bsrc=["\']([^"\']+)', webpage, 'embed url'))
+        video_id = self._match_id(embed_url)  # embed url has unique slug
+        player = self._download_webpage(embed_url, video_id, 'Downloading player page')
+        if 'MESSAGE_GEOBLOCKED' in player:
+            self.raise_geo_restricted(countries=self._GEO_COUNTRIES)
 
-        formats, subtitles = self._extract_m3u8_formats_and_subtitles(source, video_id, ext='mp4')
+        source = self._html_search_regex(r'source:\s*\'([^\']+)', player, 'source')
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(source, video_id, 'mp4')
```

Comment:
DV sources (described video) can also be extracted:
```suggestion
        source = self._html_search_regex(r'source:\s*\'([^\']+)', player, 'source')
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(source, video_id, 'mp4')
        
        if dv_source := self._html_search_regex(r'dvSource:\s*\'([^\']+)', player, 'dv source', default=None):
            fmts, subs = self._extract_m3u8_formats_and_subtitles(
                dv_source, video_id, 'mp4', m3u8_id='dv', preference=-2, fatal=False)
            formats.extend(fmts)
            self._merge_subtitles(subs, target=subtitles)
```
Here is an example: https://www.nfb.ca/film/lay-down-your-heart/